docs/gallery/general/plot_read_demo.py

```diff
@@ -0,0 +1,261 @@
+plt.imshow(image, aspect='auto')
+
+####################
+# Access single unit data
+# --------
+# Data and metadata about sorted single units are stored in :py:class:`~pynwb.misc.Units`
+# object. It stores metadata about each single unit in a tabular form, where each row represents
+# a unit with spike times and additional metadata.
+#
+# .. seealso::
+#     You can learn more about units in the :ref:`ecephys_tutorial` tutorial.
+#
+
+units = nwbfile.units
+
+####################
+# We can view the single unit data as a :py:class:`~pandas.DataFrame`.
+
+units_df = units.to_dataframe()
+print(units_df)
+
+####################
+# To access the spike times of the first single unit, index nwb.units with the column
+# name "spike_times" and then the row index, 0. All times in NWB are stored in seconds
+# relative to the session start time.
+
+print(units["spike_times"][0])
+
+####################
+# Visualize spiking activity relative to stimulus onset
+# --------
+# We can look at when these single units spike relative to when image stimuli were presented to the subject.
+# We will iterate over the first 3 units and get their spike times.
+# Then for each unit, we iterate over each stimulus onset time and compute the spike times relative
+# to stimulus onset. Finally, create a raster plot and histogram of these aligned spike times.
+
+before = 1.  # in seconds
+after = 3.
+
+# Get the stimulus times for all stimuli
+stim_on_times = stimulus_presentation.timestamps[:]
+
+for unit in range(3):
+    unit_spike_times = nwbfile.units["spike_times"][unit]
+    trial_spikes = []
+    for time in stim_on_times:
+        # Compute spike times relative to stimulus onset
+        aligned_spikes = unit_spike_times - time
+        # Keep only spike times in a given time window around the stimulus onset
+        aligned_spikes = aligned_spikes[
+            (-before < aligned_spikes) & (aligned_spikes < after)]
+        trial_spikes.append(aligned_spikes)
+    fig, axs = plt.subplots(2, 1, sharex="all")
+    plt.xlabel('time (s)')
+    axs[0].eventplot(trial_spikes)
+
+    axs[0].set_ylabel('trial')
+    axs[0].set_title('unit {}'.format(unit))
+    axs[0].axvline(0, color=[.5, .5, .5])
+
+    axs[1].hist(np.hstack(trial_spikes), 30)
+    axs[1].axvline(0, color=[.5, .5, .5])
+
+####################
+# Access Trials
+# --------
+# Trials are stored as :py:class:`~pynwb.epoch.TimeIntervals` object which is a subclass
+# of :py:class:`~hdmf.common.table.DynamicTable`. :py:class:`~hdmf.common.table.DynamicTable` objects are used to store
+# metadata about each trial in a tabular form, where each row represents a trial and has a start time, stop time,
+# and additional metadata.
+#
+# .. seealso::
+#     You can learn more about trials in the :ref:`basic_trials` tutorial section.
+#
+# Similarly to :py:class:`~pynwb.misc.Units` we can view trials as a :py:class:`pandas.DataFrame`.
+
+trials_df = nwbfile.trials.to_dataframe()
+print(trials_df)
```

Comment:
To get a prettier display of the table, you  can just place the dataframe itself as the last entry of a cell so that it will be rendered in html directly using a display that is essentially the same as how you would see the table in a Jupyter notebook. I.e., instead pf calling ``print(trials_df)`` you can simply write ``trials_df`` and sphinx gallery will render it

```suggestion
trials_df
```

Reply:
Thank you! I removed them